src/FieldWarning/Assets/Units/Component/OrderQueue/IOrder.cs

```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
+
+namespace PFW.Units.Component.OrderQueue
+{
+    public interface IOrder
+    {
+        bool OrderComplete();
+        void Deactivate();
+        void ProcessWaypoint();
+        Vector3 Destination { get; }

```

Comment:
I think with interfaces its especially important to document the intent behind the methods.